delfin/drivers/hpe/hpe_msa/ssh_handler.py

```diff
@@ -746,47 +765,56 @@ def get_storage_view_list(storage_view_list, vol_type, storage_id,
-                                                format(native_port_group_id,
-                                                       port_name)
-                                        else:
-                                            native_port_group_id = "{0}" \
-                                                .format(port_name)
-                    native_storage_host_id = None
-                    if host_list:
-                        for host_value in host_list:
-                            host_durable_id = host_value.get('description')
-                            if host_durable_id == mapped_id:
-                                native_storage_host_id = host_value.get(
-                                    'native_storage_host_id')
-                    if not native_storage_host_id:
-                        if initiators_list:
-                            for initiators in initiators_list:
-                                initiators_durable_id = initiators.get(
-                                    'native_storage_host_initiator_id')
-                                if initiators_durable_id == mapped_id:
-                                    native_storage_host_id = initiators.get(
-                                        'native_storage_host_id')
+                    native_host_group_name = 'native_storage_host_id'
+                    if '.*.*' in view_name:
+                        native_host_group_name = 'native_storage_host_group_id'
+                    native_port_group_dict = \
+                        self.get_port_group_id_and_name(port_number,
+                                                        storage_port_list)
+                    native_port_group_id = native_port_group_dict.get(
+                        'native_port_group_id')
+                    native_storage_host_id = self.get_storage_host_id(
+                        host_list, mapped_id, initiators_list,
+                        storage_host_groups, view_name)
                     view_map = {
                         "name": view_name,
                         "description": view_name,
                         "storage_id": storage_id,
                         "native_masking_view_id":
                             native_masking_view_id + volume_id,
                         native_volume_group_name: volume_id,
-                        "native_storage_host_id": native_storage_host_id
+                        native_host_group_name: native_storage_host_id
                     }
                     if native_port_group_id:
                         view_map['native_port_group_id'] = \
                             "port_group_" + native_port_group_id
                     views_list.append(view_map)
             return views_list
+
+    @staticmethod
+    def get_storage_host_id(host_list, mapped_id, initiators_list,
+                            storage_host_groups, view_name):
+        native_storage_host_id = None
+        if host_list:
```

Comment:
if 'host_list' is a list, no need to add 'if host_list'

Reply:
Delete judgment if 'storage_port_list'

Reply:
Delete judgment if 'host_list'